backend/de.metas.material/event/src/main/java/de/metas/material/event/supplyrequired/SupplyRequiredDecreasedEvent.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * #%L
+ * metasfresh-material-event
+ * %%
+ * Copyright (C) 2025 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+
+package de.metas.material.event.supplyrequired;
+
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import de.metas.material.event.MaterialEvent;
+import de.metas.material.event.commons.EventDescriptor;
+import de.metas.material.event.commons.SupplyRequiredDescriptor;
+import lombok.Builder;
+import lombok.NonNull;
+import lombok.Value;
+import lombok.extern.jackson.Jacksonized;
+import org.adempiere.util.lang.impl.TableRecordReference;
+import org.eevolution.productioncandidate.model.PPOrderCandidateId;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+
+import static de.metas.material.event.MaterialEventConstants.MD_CANDIDATE_TABLE_NAME;
+
+@Value
+@Builder
+@Jacksonized
+public class SupplyRequiredDecreasedEvent implements MaterialEvent
+{
+	public static final String TYPE = "SupplyRequiredDecreasedEvent";
+
+	@NonNull SupplyRequiredDescriptor supplyRequiredDescriptor;
+	@NonNull Collection<PPOrderCandidateId> ppOrderCandidateIds;
+	@NonNull Collection<Integer> ddOrderCandidateIds;
+	@NonNull Collection<Integer> purchaseCandidateIds;
```

Comment:
why do we have to provide the ppOrderCandidateIds, ddOrderCandidateIds and purchaseCandidateIds?

IMHO, figuring out what to do and which are the candidates to adjust in case of decreased event it's the job of the handler.

Reply:
cleaned up those fields